version: '3'

services:
  bitwarden:
    image: bitwardenrs/server:latest #swap tag to raspberry to run on a raspberry pi <- Not neccessary due to multiarch support from 2021
    expose:
      - "80"
      - "3012"
    volumes:
      - ./data/bitwarden:/data
    restart: on-failure
    environment:
      WEBSOCKET_ENABLED: 'true'
 
  nginx:
    image: nginx:stable-alpine 	#1.15-alpine				# Why use old image? changed to newest version
    restart: always
    ports:
      - "80:80"								#Check up on the port listens
      - "443:443"
    volumes:
      - ./data/nginx:/etc/nginx
      - ./data/ssl:/etc/ssl

  bw_backup:
    image: bruceforce/bw_backup:latest					# Multi-arch support in latest release - > Not neccessarry #swap tag to rpi3 to run on a raspberry pi
    container_name: bw_backup
    restart: on-failure
    depends_on:
      - bitwarden
    volumes:
      - ./data/bitwarden:/data
      - /opt/Bitwarden-Backup:/backup
      - /etc/timezone:/etc/timezone:ro
      - /etc/localtime:/etc/localtime:ro
    environment:
      - DB_FILE=/data/db.sqlite3
      - BACKUP_FILE=/backup/backup.sqlite3
      - CRON_TIME=0 * * * *
      - TIMESTAMP=false
      - UID=0
      - GID=0

  bw_backup_usb:
    image: bruceforce/bw_backup:latest					# Multi-arch support in latest release - > Not neccessarry #swap tag to rpi3 to run on a raspberry pi
    container_name: bw_backup_usb
    restart: on-failure
    depends_on:
      - bitwarden
    volumes:
      - ./data/bitwarden:/data                  # Where the backup data is located -> No need to change 
      - /media/usb-backup:/backup               # Where the backup is sent to -> I want to sync with USB-drive
      - /etc/timezone:/etc/timezone:ro
      - /etc/localtime:/etc/localtime:ro
    environment:
      - DB_FILE=/data/db.sqlite3
      - BACKUP_FILE=/backup/backup.sqlite3
      - CRON_TIME=0 12 * * *                    # Run once a day at midday.
      - TIMESTAMP=True
      - UID=0
      - GID=0

